src/client/components/draft/paragraph/utils/convert.js

```diff
@@ -0,0 +1,152 @@
+import React from 'react'
+import { convertFromHTML, convertToHTML } from 'draft-convert'
+import { stripGoogleStyles } from 'client/components/rich_text/utils/text_stripping'
+import { styleNamesFromMap, styleNodesFromMap } from './utils'
+
+/**
+ * Helpers for draft-js Paragraph component data conversion
+*/
+
+export const draftDefaultStyles = [
+  'BOLD',
+  'CODE',
+  'ITALIC',
+  'STRIKETHROUGH',
+  'UNDERLINE'
+]
+
+export const convertHtmlToDraft = (html, hasLinks, allowedStyles) => {
+  /**
+   * Convert HTML to Draft ContentState
+   */
```

Comment:
For these doc-blocks to be parsed correctly by tooling they need to idiomatically exist on the outside of the function:

```js
/**
 * Convert HTML to Draft ContentState
 */
export const convertHtmlToDraft = (html, hasLinks, allowedStyles) => {
```

Reply:
They seem to be working fine locally, what kind of parsing errors would you expect? Is it possible this is why circle is failing?

Reply:
still interested to know why this errors, don't see any obvious flags from a quick google

Reply:
It's not an error in the sense that the code will break; it's that editor tooling follows idiomatic patterns of use based on a spec. Since docblocking is top down (documentation, then code) and thus meant to be placed on the outside of the function, when placed on the inside the assigned scope becomes invalid (in this case appearing for `let cleanedHtml = stripGoogleStyles(html)`) 

You can test this by placing your cursor above the function and typing `/**` and hitting enter. It will fill in placeholders for variables and params. If on the inside the parser no longer knows how to capture context, and tooling that reads dockblocks will link to an incorrect line number.

Reply:
Good point @damassi. Yeah @eessex we might as well follow the JSDoc format for documenting functions. That way it gets picked up by Typescript and consequently VSCode.

See: http://usejsdoc.org/about-getting-started.html
https://github.com/artsy/reaction/blob/master/src/Components/Authentication/Types.ts#L35

Reply:
Visually I do like the inner Python-style comment-blocks as it looks a bit cleaner but that format isn't really seen in JSland.

One last thought is that all of these files could be `.ts` since positron now supports TypeScript. Not suggesting that everything needs to be properly typed out with interfaces, but basic out-of-the-box checks are still safer than `.js`.